Mesh_3/demo/Mesh_3/Volume_plane_thread.h

```diff
@@ -10,12 +10,13 @@
 #include <QThread>
 #include <vector>
 
+template<typename Word>
 struct Clamp_to_one_zero_range {
-  std::pair<float, float> min_max;
-  float operator()(const float& inVal) {
-    float inValNorm = inVal - min_max.first;
-    float aUpperNorm = min_max.second - min_max.first;
-    float bValNorm = inValNorm / aUpperNorm;
+  std::pair<Word, Word> min_max;
+  Word operator()(const Word& inVal) {
+    Word inValNorm = inVal - min_max.first;
+    Word aUpperNorm = min_max.second - min_max.first;
+    Word bValNorm = inValNorm / aUpperNorm;
```

Comment:
`Word` can be an integral type? I wrote this code under the assumption that it would always be a floating point type, because I was blissfully ignorant of what `CGAL_IMAGE_IO_CASE` actually does. Then that code can't possibly work. I'll revert that part of the change and add a static cast.